fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingMutuallyByRecords.java

```diff
@@ -514,6 +536,59 @@ private void infiniteLoopProtection(final Range range, final List<Range> missing
         }
     }
 
+    // anyJumper:
+    //  During the ANY iteration, we wish to avoid cases of two indexers competing on the same fragment.
+    //  It is done by recording the exception and the fragment info, then - during the next iteration:
+    //      If at the same fragment, same range: re-throw
+    //      If at the same fragment, different range: jump to the next fragment
+    boolean anyJumperSaysJump(Range rangeToBuild) {
+        if (anyJumperEx == null || fragmentIterationType != FragmentIterationType.ANY) {
+            return false;
+        }
+        if (LOGGER.isInfoEnabled()) {
+            LOGGER.info(KeyValueLogMessage.build("anyJumper: check if should jump",
+                            "anyJumperRange", anyJumperRange,
+                            "anyJumperCurrent", anyJumperCurrent,
+                            "anyJumperEx", anyJumperEx,
+                            LogMessageKeys.RANGE, rangeToBuild)
+                    .addKeysAndValues(fragmentLogMessageKeyValues())
+                    .toString());
+        }
+        if (anyJumperCurrent != fragmentCurrent) {
+            // Here: this way or another, the conflicting fragment is fully built now. Irrelevant.
+            anyJumperEx = null;
+            return false;
+        }
+        if (anyJumperRange.equals(rangeToBuild)) {
+            // Here: in hindsight, this exception was not caused by a rangeSet conflict. Rethrow it.
```

Comment:
Is this because, if the conflict where a rangeSet conflict, on retry, we would have gotten a different `rangeToBuild`, so if we got this, it means we conflicted on something else?

Reply:
Yes.